OpenTerm/Util/Parsing & Formatting/Parser.swift

```diff
@@ -116,44 +142,41 @@ class Parser {
 			buffer = Data()
 		}
 
-		return (str, didEnd)
+		return didEnd
 	}
 
 	/// Decode UTF-8 string from the given data.
 	/// This is a custom implementation that decodes what characters it can then returns whatever it can't,
 	/// which is necessary since data can come in arbitrarily-sized chunks of bytes, with characters split
 	/// across multiple chunks.
 	/// The first time decoding fails, all of the rest of the data will be returned.
-	private func decodeUTF8(fromData data: Data) -> (decoded: NSAttributedString, remaining: Data, didEnd: Bool) {
+	private func decodeUTF8(fromData data: Data) -> (remaining: Data, didEnd: Bool) {
 		let byteArray = [UInt8](data)
 
 		var utf8Decoder = UTF8()
-		let str = NSMutableAttributedString()
 		var byteIterator = byteArray.makeIterator()
 		var decodedByteCount = 0
 		var didEnd: Bool = false
 		Decode: while !didEnd {
 			switch utf8Decoder.decode(&byteIterator) {
 			case .scalarValue(let v):
-				var output: NSAttributedString? = nil
-				(output, didEnd) = self.handle(Character(v))
-				if let output = output {
-					str.append(output)
-				}
+				didEnd = self.handle(Character(v))
 				decodedByteCount += UTF8.encode(v)!.count
 			case .emptyInput, .error:
 				break Decode
 			}
 		}
 
 		let remaining = Data.init(bytes: byteArray.suffix(from: decodedByteCount))
-		return (str, remaining, didEnd)
+		return (remaining, didEnd)
 	}
 
 	/// This method is called for each UTF-8 character that is received.
 	/// It should perform state changes based on that character, then
 	/// return an attributed string that renders the character
-	private func handle(_ character: Character) -> (output: NSAttributedString?, didEnd: Bool) {
+	//
+	// swiftlint:disable cyclomatic_complexity
```

Comment:
I decided to disable the swiftlint error in this method instead of splitting it up into smaller methods. This method basically models a state machine, and doesn't do much other than change states and call delegate methods. It would, in my opinion, reduce readability a lot if I were to move the inner switches into methods and call those.

@louisdh thoughts?

Reply:
That's fine.